politeiawww/proposals.go

```diff
@@ -625,62 +625,71 @@ func (p *politeiawww) processBatchVoteSummary(bvs www.BatchVoteSummary) (*www.Ba
 	return &res, nil
 }
 
+func convertVoteErrorCodeToWWW(errcode ticketvote.VoteErrorT) decredplugin.ErrorStatusT {
+	switch errcode {
+	case ticketvote.VoteErrorInvalid:
+		return decredplugin.ErrorStatusInvalid
+	case ticketvote.VoteErrorInternalError:
+		return decredplugin.ErrorStatusInternalError
+	case ticketvote.VoteErrorRecordNotFound:
+		return decredplugin.ErrorStatusProposalNotFound
+	case ticketvote.VoteErrorVoteBitInvalid:
+		return decredplugin.ErrorStatusInvalidVoteBit
+	case ticketvote.VoteErrorVoteStatusInvalid:
+		return decredplugin.ErrorStatusVoteHasEnded
+	case ticketvote.VoteErrorTicketAlreadyVoted:
+		return decredplugin.ErrorStatusDuplicateVote
+	case ticketvote.VoteErrorTicketNotEligible:
+		return decredplugin.ErrorStatusIneligibleTicket
+	default:
+		return decredplugin.ErrorStatusInternalError
+	}
+}
+
 func (p *politeiawww) processCastVotes(ballot *www.Ballot) (*www.BallotReply, error) {
 	log.Tracef("processCastVotes")
 
-	challenge, err := util.Random(pd.ChallengeSize)
-	if err != nil {
-		return nil, err
-	}
-
-	// Verify proposal tokens
+	// Prep plugin command
+	bp := ticketvote.Ballot{}
```

Comment:
When initializing something to its zero value, use the `var bp ticketvote.Ballot` notation.